dotnet/src/Connectors/Connectors.AI.PaLM/TextCompletion/PaLMTextCompletion.cs

```diff
@@ -0,0 +1,234 @@
+    }
+
+    /// <inheritdoc/>
+    public async IAsyncEnumerable<ITextStreamingResult> GetStreamingCompletionsAsync(
+        string text,
+        CompleteRequestSettings requestSettings,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        foreach (var completion in await this.ExecuteGetCompletionsAsync(text, cancellationToken).ConfigureAwait(false))
+        {
+            yield return completion;
+        }
+    }
+
+    /// <inheritdoc/>
+    public async Task<IReadOnlyList<ITextResult>> GetCompletionsAsync(
+        string text,
+        CompleteRequestSettings requestSettings,
+        CancellationToken cancellationToken = default)
+    {
+        return await this.ExecuteGetCompletionsAsync(text, cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    [Obsolete("This method is deprecated and will be removed in one of the next SK SDK versions.")]
+    public void Dispose()
+    {
+        if (this._disposeHttpClient)
+        {
+            this._httpClient.Dispose();
+        }
+    }
+
+    #region private ================================================================================
+
+    private async Task<IReadOnlyList<ITextStreamingResult>> ExecuteGetCompletionsAsync(string text, CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            var completionRequest = new TextCompletionRequest();
+            completionRequest.prompt.text = text;
+
+            using var httpRequestMessage = new HttpRequestMessage()
+            {
+                Method = HttpMethod.Post,
+                RequestUri = this.GetRequestUri(),
+                Content = new StringContent(JsonSerializer.Serialize(completionRequest)),
+            };
+
+            httpRequestMessage.Headers.Add("User-Agent", HttpUserAgent);
+            
+            //if (!string.IsNullOrEmpty(this._apiKey))
+            //{
+            //    httpRequestMessage.Headers.Add("Authorization", $"Bearer {this._apiKey}");
+            //}
+
+            using var response = await this._httpClient.SendAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+            response.EnsureSuccessStatusCode();
+
+            var body = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+
+            TextCompletionResponse? completionResponse = JsonSerializer.Deserialize<TextCompletionResponse>(body);
+
+            if (completionResponse is null)
+            {
+                throw new AIException(AIException.ErrorCodes.InvalidResponseContent, "Unexpected response from model")
+                {
+                    Data = { { "ResponseData", body } },
+                };
+            }
+
+            if (completionResponse.candidates is null)
+            {
+                completionResponse = new TextCompletionResponse() { candidates = new Candidate[] { new Candidate() { output = "response is null, please try another input." } } };
```

Comment:
Is the response without candidates useful for calling code? Isn't it a violation of the method contract? Shouldn't an exception be thrown in such cases instead?

Reply:
it's came from the API itself, it will not answer some questions if violate their rules. it will send different Json schema, I have added another schema to handle this. fixed